pkg/checkup/workspace.go

```diff
@@ -371,3 +371,26 @@ func (w *workspace) deleteNamespace(client *kubernetes.Clientset) error {
 
 	return nil
 }
+
+func (w *workspace) RetrieveCheckupStatus(client *kubernetes.Clientset) (*Status, error) {
+	resultsConfigMap, err := w.getResultsConfigMap(client)
+	if err != nil {
+		return nil, err
+	}
+
+	status, err := newStatusFromConfigMap(resultsConfigMap)
+	if err != nil {
+		return nil, err
+	}
+
+	return status, nil
+}
+
+func (w *workspace) getResultsConfigMap(client *kubernetes.Clientset) (*corev1.ConfigMap, error) {
```

Comment:
Please avoid defining the return value at the signature, in general its harder to follow.
There are use cases for using it, as far as I remember when working with callbacks,
but not on common case.

Reply:
Can you please elaborate?